apps/etterlatte-beregning/src/main/kotlin/avkorting/AarligInntektsjusteringService.kt

```diff
@@ -0,0 +1,60 @@
+package no.nav.etterlatte.avkorting
+
+import no.nav.etterlatte.libs.common.beregning.AarligInntektsjusteringAvkortingSjekkRequest
+import no.nav.etterlatte.libs.common.beregning.AarligInntektsjusteringAvkortingSjekkResponse
+import no.nav.etterlatte.libs.common.beregning.AvkortingGrunnlagLagreDto
+import no.nav.etterlatte.libs.ktor.token.BrukerTokenInfo
+import no.nav.etterlatte.sanksjon.SanksjonService
+import org.slf4j.LoggerFactory
+import java.time.YearMonth
+import java.util.UUID
+
+class AarligInntektsjusteringService(
+    private val avkortingService: AvkortingService,
+    private val avkortingRepository: AvkortingRepository,
+    private val sanksjonService: SanksjonService,
+) {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    fun hentSjekkAarligInntektsjustering(
+        request: AarligInntektsjusteringAvkortingSjekkRequest,
+    ): AarligInntektsjusteringAvkortingSjekkResponse {
+        val sanksjoner = sanksjonService.hentSanksjon(request.sisteBehandling)
+        return AarligInntektsjusteringAvkortingSjekkResponse(
+            sakId = request.sakId,
+            aar = request.aar,
+            harInntektForAar = avkortingRepository.harSakInntektForAar(request),
+            harSanksjon = sanksjoner?.any { it.tom == null } ?: false,
+        )
+    }
+
+    suspend fun kopierAarligInntektsjustering(
+        behandlingId: UUID,
+        forrigeBehandlingId: UUID,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Avkorting {
+        avkortingService.tilstandssjekk(behandlingId, brukerTokenInfo)
+        logger.info("Oppretter avkorting for nytt inntektsår med siste inntekt fra behandling=$forrigeBehandlingId")
+        val forrigeAvkorting = avkortingService.hentForrigeAvkorting(forrigeBehandlingId)
+
+        val siseInntekt =
+            forrigeAvkorting.aarsoppgjoer
+                .last()
+                .inntektsavkorting
+                .last()
+                .grunnlag
```

Comment:
Det er mulig dette fikses lengre nede -- men hadde det vært naturlig å hatt en sjekk på at forrige grunnlag vi henter ut er fra i år? Hvis vi plutselig nå prøver å legge inn grunnlag for året etter neste år hadde jeg likt en treffende feilmelding herifra først :p